From ceb8cebf040fc3a04724e15ebd4d135ab42107c8 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Sun, 13 Aug 2023 14:24:37 +0200
Subject: [PATCH] CMakeLists.txt: use systemdsystemunitdir instead of
 /lib/systemd/system

:Release Notes:
There isn't a better WEBOS_INSTALL_* variable for /lib/systemd/system
but with usrmerge systemd_system_unitdir is /usr/lib/systemd/system
not /lib/systemd/system:

:Detailed Notes:
  # $systemd_system_unitdir [2 operations]
  #   exported /OE/build/luneos-nanbield/openembedded-core/meta/conf/bitbake.conf:39
  #     [export] "1"
  #   set /OE/build/luneos-nanbield/openembedded-core/meta/conf/bitbake.conf:39
  #     "${nonarch_base_libdir}/systemd/system"
  # pre-expansion value:
  #   "${nonarch_base_libdir}/systemd/system"
  export systemd_system_unitdir="/usr/lib/systemd/system"

Fixes:
http://gecko.lge.com:8000/Errors/Details/663214

ERROR: Didn't find service unit 'contextintentmgr.service', specified in SYSTEMD_SERVICE:com.webos.service.contextintentmgr.

We also cannot use WEBOS_INSTALL_BASE_LIBDIR, because we really need
noarch base_libdir, otherwise in multilib the aarch64 build fails again
because the service file will be in:
com.webos.service.contextintentmgr/1.0.0-11-r5/image/lib64/systemd/system/contextintentmgr.service

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-24156] CCC: Various build fixes

Change-Id: Id15048d0f0f186578b50718c1656f2c5125b4790
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/com.webos.service.contextintentmgr/+/364504 CMakeLists.txt: use systemdsystemunitdir instead of /lib/systemd/system]

 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad59e86..317aaf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,6 @@ install(DIRECTORY . DESTINATION ${INSTALL_DIR}
         PATTERN "tempdb" EXCLUDE
         PATTERN "*.qmlproject" EXCLUDE
         PATTERN "README.md" EXCLUDE)
-install(DIRECTORY files/systemd/services/ DESTINATION /lib/systemd/system/)
+install(DIRECTORY files/systemd/services/ DESTINATION ${systemdsystemunitdir})
 
 webos_build_system_bus_files()
